crates/turbopack-node/src/render/render_static.rs

```diff
@@ -201,3 +181,176 @@ async fn static_error(
+#[turbo_tasks::value(eq = "manual", cell = "new", serialization = "none")]
+pub struct RenderStreamSender {
+    #[turbo_tasks(trace_ignore, debug_ignore)]
+    get: Box<dyn Fn() -> UnboundedSender<RenderItemResult> + Send + Sync>,
+}
+
+#[turbo_tasks::value(transparent, eq = "manual", cell = "new", serialization = "none")]
+struct RenderStream(#[turbo_tasks(trace_ignore)] Stream<RenderItemResult>);
+
+#[turbo_tasks::function]
+fn render_stream(
+    pool: NodeJsPoolVc,
+    data: RenderDataVc,
+    intermediate_asset: AssetVc,
+    intermediate_output_path: FileSystemPathVc,
+    project_dir: FileSystemPathVc,
+    error_path: FileSystemPathVc,
+    fallback_page: DevHtmlAssetVc,
+) -> RenderStreamVc {
+    // Note the following code uses some hacks to create a child task that produces
+    // a stream that is returned by this task.
+
+    // We create a new cell in this task, which will be updated from the
+    // [render_stream_internal] task.
+    let cell = turbo_tasks::macro_helpers::find_cell_by_type(*RENDERSTREAM_VALUE_TYPE_ID);
+
+    // We initialize the cell with a stream that is open, but has no values.
+    // The first [render_stream_internal] pipe call will pick up that stream.
+    let (sender, receiver) = unbounded();
+    cell.update_shared(RenderStream(Stream::new_open(vec![], Box::new(receiver))));
+    let initial = Mutex::new(Some(sender));
+
+    // run the evaluation as side effect
+    render_stream_internal(
+        pool,
+        data,
+        intermediate_asset,
+        intermediate_output_path,
+        project_dir,
+        error_path,
+        fallback_page,
+        RenderStreamSender {
+            get: Box::new(move || {
+                if let Some(sender) = initial.lock().take() {
+                    sender
+                } else {
+                    // In cases when only [render_stream_internal] is (re)executed, we need to
+                    // update the old stream with a new value.
+                    let (sender, receiver) = unbounded();
+                    cell.update_shared(RenderStream(Stream::new_open(vec![], Box::new(receiver))));
+                    sender
+                }
+            }),
+        }
+        .cell(),
+    );
+
+    let raw: RawVc = cell.into();
+    raw.into()
+}
+
+#[turbo_tasks::function]
+async fn render_stream_internal(
+    pool: NodeJsPoolVc,
+    data: RenderDataVc,
+    intermediate_asset: AssetVc,
+    intermediate_output_path: FileSystemPathVc,
+    project_dir: FileSystemPathVc,
+    error_path: FileSystemPathVc,
+    fallback_page: DevHtmlAssetVc,
+    sender: RenderStreamSenderVc,
+) {
+    mark_finished();
+    let Ok(sender) = sender.await else {
+        // Impossible to handle the error in a good way.
+        return;
+    };
+
+    let stream = generator! {
+        let data = data.await?;
+        // Read this strongly consistent, since we don't want to run inconsistent
+        // node.js code.
+        let pool = pool.strongly_consistent().await?;
```

Comment:
This strongly_consistent will no longer work since the pool is an argument and not a function call.
We need to move the get_pool into this method.
Probably the same for the evaluate pool

Reply:
Done.